### DISCUSSION TITLE
An error occurs when logging in using the chrome plug-in: Cannot read properties of nul(reading ‘iterations‘)

### DISCUSSION THREAD

**Discussion by lonely06** [dani-garcia/vaultwarden#4010] [Q&A]
I pulled the docker version of "vaultwarden/server:latest" and could log in normally using the web page. However, when using the edge/chrome extension, an error "Cannot read properties of nul(reading 'iterations‘)" appeared.

**Comment by BlackDex** [comment]
What do the Vaultwarden logs report during the login action that falls?

**Reply by lonely06** [reply]
[2023-10-30 06:53:55.115][request][INFO] GET /api/config
[2023-10-30 06:53:55.115][response][INFO] GET /<p..> [10] (web_files) => 404 Not Found
[2023-10-30 06:54:03.665][request][INFO] GET /api/devices/knowndevice
[2023-10-30 06:54:03.665][response][INFO] GET /<p..> [10] (web_files) => 404 Not Found
[2023-10-30 06:54:15.492][request][INFO] POST /identity/accounts/prelogin
[2023-10-30 06:54:15.493][response][INFO] 404 Not Found
[2023-10-30 06:54:53.575][request][INFO] POST /identity/accounts/prelogin
[2023-10-30 06:54:53.575][response][INFO] 404 Not Found
[2023-10-30 06:54:55.556][request][INFO] POST /identity/accounts/prelogin
[2023-10-30 06:54:55.556][response][INFO] 404 Not Found

**Reply by lonely06** [reply]
This will appear every time you click the login button:
[2023-10-30 06:54:55.556][request][INFO] POST /identity/accounts/prelogin
[2023-10-30 06:54:55.556][response][INFO] 404 Not Found

**Reply by BlackDex** [reply]
You probably have a very old version of Vaultwarden.
Check `/admin/diagnostics`

**Reply by lonely06** [reply]
My current Vaultwarden version is: 2.25.0. How do I go about checking `/admin/diagnostics` ?

**Reply by BlackDex** [reply]
So, something is out of sync, or your `pull` worked, but you didn't run `up -d`.

**Reply by lonely06** [reply]
This is my command to create the container: `docker run -d --name vaultwarden -v /home/bitwarden/:/data/ -e SIGNUPS_ALLOWED=false -p 8033:80 docker.io/vaultwarden/server:latest`

**Reply by BlackDex** [reply]
Yes, but, if you pull a new image version, it will not update this when you stop and start the container again.
You need to remove the container and re-create it with the same config options as you did before.

See: https://github.com/dani-garcia/vaultwarden/wiki/Updating-the-vaultwarden-image

**Reply by lonely06** [reply]
I stopped and deleted the current container, then deleted the local mapping file and reused `docker run -d --name vaultwarden -v /home/bitwarden/:/data/ -e SIGNUPS_ALLOWED=false -p 8033:80 docker .io/vaultwarden/server:latest` command was created. But it still appears:
[2023-10-30 14:09:39.826][request][INFO] POST /identity/accounts/prelogin
[2023-10-30 14:09:39.826][response][INFO] 404 Not Found

**Reply by BlackDex** [reply]
And what does `docker exec -it vaultwarden /vaultwarden --version` report?

**Reply by lonely06** [reply]
show `vaultwarden 1.23.1`

**Reply by lonely06** [reply]
Thank you very much for your advice, this problem has been successfully solved. It's because the version pulled using `docker:latest` is 1.23.1. I pulled the latest version through `docker:1.29.2` and successfully solved this error. But I have a question why `docker:latest` does not pull the latest version?

**Reply by BlackDex** [reply]
It should. You are sure you run `docker pull vaultwarden/server:latest`?

Because that should pull in the new version.

**Reply by lonely06** [reply]
Yes, I confirm. I installed docker-ce on a new centos 7 system, used `docker pull vaultwarden/server:latest` to pull, and then used `docker run -d --name vaultwarden -v /home/bitwarden/:/data/ - e SIGNUPS_ALLOWED=false -p 8033:80 docker.io/vaultwarden/server:latest` for deployment. After entering `docker exec -it vaultwarden /vaultwarden --version`, vaultwarden 1.23.1 is displayed.

This is the entire process:
```
[root@localhost ~]# sudo systemctl restart docker
[root@localhost ~]# docker pull vaultwarden/server:latest
latest: Pulling from vaultwarden/server
ffbb094f4f9e: Pull complete
3953aa1aac1b: Pull complete
f94ad8d4973a: Pull complete
31d34e7d15a6: Pull complete
e436b51dcd0d: Pull complete
8fdbf8c74aad: Pull complete
c79b268f4fca: Pull complete
Digest: sha256:08d8edab2311ee36de9f1e2dd7f6ae18680a7fcbf00bab98d63fd4ff82b285e2
Status: Downloaded newer image for vaultwarden/server:latest
docker.io/vaultwarden/server:latest
[root@localhost ~]# docker run -d --name vaultwarden -v /home/bitwarden/:/data/ -e SIGNUPS_ALLOWED=false -p 8033:80 docker.io/vaultwarden/server:latest
334496d68da64df4b8527e2cd50f17d42763f7bfddfd001a9022a2f4f017055f
[root@localhost ~]# docker exec -it vaultwarden /vaultwarden --version
vaultwarden 1.23.1
```

**Reply by BlackDex** [reply]
Strange, for me it works just fine:

```
docker run --rm -it docker.io/vaultwarden/server:latest /vaultwarden --version
Unable to find image 'vaultwarden/server:latest' locally
latest: Pulling from vaultwarden/server
52d2b7f179e3: Pull complete 
22156f0dbb04: Pull complete 
e74358c75985: Pull complete 
add9a3825a78: Pull complete 
2c5b216fc04d: Pull complete 
7d794a488eb9: Pull complete 
Digest: sha256:960d4b59eef0208808d21cb25f1bc7296465dd1080a41fcb93b476d4625c8e5a
Status: Downloaded newer image for vaultwarden/server:latest
vaultwarden 1.29.2
```

The same for `ghcr.io/dani-garcia/vaultwarden:latest` and `quay.io/vaultwarden/server:latest`.
I would guess something in  your docker cache is not going well then.

**Reply by BlackDex** [reply]
Maybe you use a mirror which doesn't provide the right version?

**Reply by lonely06** [reply]
It may be a problem with the mirror source.
I set up the image source before installing docker-ce:
```
yum-config-manager \
    --add-repo \
    https://mirrors.aliyun.com/docker-ce/linux/centos/docker-ce.repo
    
sed -i 's/download.docker.com/mirrors.aliyun.com\/docker-ce/g' /etc/yum.repos.d/docker-ce.repo

yum makecache fast
```
After the installation is complete, Alibaba Cloud's image acceleration service is set up:
```
sudo mkdir -p /etc/docker
sudo tee /etc/docker/daemon.json <<-'EOF'
{
  "registry-mirrors": ["https://vc4th1n2.mirror.aliyuncs.com"]
}
EOF
sudo systemctl daemon-reload
sudo systemctl restart docker
```

Is it because of this problem?

**Reply by BlackDex** [reply]
Probably yes, they do not have a `latest` tagged version of Vaultwarden which has the actual latest version of Vaultwarden.
That is not something we can fix.

**Reply by lonely06** [reply]
Got it, thank you again for your patient answer
